crates/bevy_render/src/render_asset.rs

```diff
@@ -120,15 +119,41 @@ impl<A: RenderAsset> Default for ExtractedAssets<A> {
 
 /// Stores all GPU representations ([`RenderAsset::PreparedAssets`](RenderAsset::PreparedAsset))
 /// of [`RenderAssets`](RenderAsset) as long as they exist.
-#[derive(Resource, Deref, DerefMut)]
-pub struct RenderAssets<A: RenderAsset>(HashMap<Handle<A>, A::PreparedAsset>);
+#[derive(Resource)]
+pub struct RenderAssets<A: RenderAsset>(HashMap<AssetId<A>, A::PreparedAsset>);
 
 impl<A: RenderAsset> Default for RenderAssets<A> {
     fn default() -> Self {
         Self(Default::default())
     }
 }
 
+impl<A: RenderAsset> RenderAssets<A> {
+    pub fn get(&self, id: impl Into<AssetId<A>>) -> Option<&A::PreparedAsset> {
+        self.0.get(&id.into())
+    }
+
+    pub fn get_mut(&mut self, id: impl Into<AssetId<A>>) -> Option<&mut A::PreparedAsset> {
+        self.0.get_mut(&id.into())
+    }
+
+    pub fn insert(
+        &mut self,
+        id: impl Into<AssetId<A>>,
+        value: A::PreparedAsset,
+    ) -> Option<A::PreparedAsset> {
+        self.0.insert(id.into(), value)
+    }
+
+    pub fn remove(&mut self, id: impl Into<AssetId<A>>) -> Option<A::PreparedAsset> {
+        self.0.remove(&id.into())
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = (AssetId<A>, &A::PreparedAsset)> {
+        self.0.iter().map(|(k, v)| (*k, v))
+    }
```

Comment:
Would an `iter_mut` be useful ? (as I see `get_mut()`)

Reply:
Added!